### PR TITLE
Refactor of Error instance hierarchy

### LINKED ISSUE #369: Inconsistency in tofu.syntax.handle
Having some `Handle[F, E]`, I can handle errors of type `E1 <: E` with the `handleWith` method, but not with `handle`: https://scastie.scala-lang.org/jzaAE02dRwuYlnD7nLaQ7Q

This is because the `handleWith` method requires `Handle[F, E1]`, which can be derived from `Handle[F, E]` using `handleDowncast`. But the `handle` method requires `HandleTo[F, F, E1]`, which can not be derived.

I think the easiest way to fix this is to require `Handle[F, E]` in the `handle` method. But I don't know if there might be cases where `HandleTo[F, F, E]` is available, but `Handle[F, E]` is not.

### PR DESCRIPTION
**Pull request by Odomontois**
Special thanks to @neko-kai for teaching me basic scala

fixes #369

### CODE REVIEW

File: core/src/main/scala/tofu/Errors.scala [hunk 1]
```diff
@@ -8,12 +8,18 @@ import tofu.internal.{CachedMatcher, DataEffectComp}
 import tofu.lift.Lift
 import tofu.optics.PUpcast.GenericSubtypeImpl
 import tofu.optics.{Downcast, Subset, Upcast}
-
-trait Raise[F[_], E] extends Raise.ContravariantRaise[F, E] {
+import scala.annotation.implicitNotFound
+
+/**
+  * Having this means a possibity of raising `E` inside type `F`
```
Comment (oskin1) [inline core/src/main/scala/tofu/Errors.scala:14]: Why not simplifying the sentence?
```suggestion
  * Allows to raise `E` inside type `F`.
```

Comment (oskin1) [reply core/src/main/scala/tofu/Errors.scala:14]: Same for other similar comments below.

Comment (Odomontois) [reply core/src/main/scala/tofu/Errors.scala:14]: Can merge all of them, if you do multisuggestion

File: core/src/main/scala/tofu/Errors.scala [hunk 2]
```diff
@@ -58,8 +65,11 @@ trait HandleTo[F[_], G[_], E] extends RestoreTo[F, G] {
     handleWith(fa)(e => G.pure(f(e)))
 }
 
-object HandleTo extends ErrorsToInstanceChain[HandleTo]
-
+/**
+  * Having this means you can recover after error of type ${E} in the ${F}
```
Comment (oskin1) [inline core/src/main/scala/tofu/Errors.scala:69]:
```suggestion
  * Allows to recover after an error of type ${E} in a ${F}.
```

File: core/src/main/scala/tofu/Errors.scala [hunk 3]
```diff
@@ -95,46 +105,57 @@ object Handle extends HandleInstances with DataEffectComp[Handle] {
...
 }
 
-sealed class HandleInstances extends ErrorsInstanceChain[Handle] {
-  final implicit def handleDowncast[F[_], E, E1](implicit h: Handle[F, E], prism: Downcast[E, E1]): Handle[F, E1] =
-    new FromPrism[F, E, E1, Handle, Downcast] with HandlePrism[F, E, E1]
-}
-
+/**
+  * Having this means you can throw and handle errors of type ${E} in the ${F} having ${G} as recover result
+  * `G` can be same as `F` or some "subconstructor" having less errors semantically
```
Comment (oskin1) [inline core/src/main/scala/tofu/Errors.scala:110]:
```suggestion
  * Allows to throw and handle errors of type ${E} in a ${F} transiting to a ${G} when recovering.
  * A `G` can either be the same as `F` or some "subconstructor" having less errors semantically.
```

File: core/src/main/scala/tofu/Errors.scala [hunk 4]
```diff
@@ -95,46 +105,57 @@ object Handle extends HandleInstances with DataEffectComp[Handle] {
...
-
-  final implicit def optionTIntance[F[_]](implicit F: Monad[F]): TC[OptionT[F, *], F, Unit] = new OptionTErrorsTo[F]
-
-  final implicit def eitherIntance[E]: TC[Either[E, *], Id, E] =
-    new EitherErrorsTo[E]
-  final implicit val optionTIntance: TC[Option, Id, Unit]      = OptionErrorsTo
-}
-
+/**
+  * Having this means you can throw and handle errors of type ${E} in the ${F}
```
Comment (oskin1) [inline core/src/main/scala/tofu/Errors.scala:117]:
```suggestion
  * Allows to throw and handle errors of type ${E} in a ${F}.
```

**Review by oskin1 — APPROVED** [review]

**Pull request merged**